### PR TITLE
replaced getProfile with getUserInfo and added redirect

### PR DESCRIPTION
**Pull request by andreadelrio**
- User is now redirected back to the page where they were before logging in.

- Replaced deprecated getProfile with getUserInfo.

- Since we are using the option redirectUrl for Auth0, there's no way to get rid of the code in index.js at this point.  (See: https://auth0.com/forum/t/react-with-react-router-redirects-before-token-is-set/3495/4)

### CODE REVIEW

File: src/routes.js [hunk 1]
```diff
@@ -45,6 +46,13 @@ var getFirstBrowserLanguage = function () {
   return null
 }
 
+var onEnter = function() {
+  let parsedParams = queryString.parse(window.location.hash)
+  let path = JSON.parse(JSON.stringify(parsedParams.state)).slice(13, -2)
```
Comment (davidascher) [inline src/routes.js]: These numbers feel magic -- can you either add a comment explaining why, and/or (preferably) do some parsing which is more robust?  I fear that this code could break mysteriously.

File: src/routes.js [hunk 2]
```diff
@@ -45,6 +46,13 @@ var getFirstBrowserLanguage = function () {
   return null
 }
 
+var onEnter = function() {
```
Comment (davidascher) [inline src/routes.js]: `onEnter` feels like the wrong name for this function. Maybe `onRedirect`?

File: src/actions.js [hunk 3]
```diff
@@ -236,4 +214,4 @@ export function search (query, selectedCategory, sortingMethod) {
         console.log('got an error after performSearch', err)
       })
   }
-}
+}
```
Comment (davidascher) [inline src/actions.js]: dropped final EOL, should be kept.

**Comment by andreadelrio** [comment]
@davidascher just made all your suggested changes.

File: src/actions.js [hunk 4]
```diff
@@ -23,54 +23,32 @@ import api from './utils/api'
...
-  return dispatch => {
-    lock.show({auth: { params: { scope: 'openid email read:users update:users update:users_app_metadata user_metadata app_metadata' }}}, (err, profile, token) => {
-      // TODO #45 in actions.js figure out why the auth promise never gets called 
-      // TODO when auth promise code fixed, remove code in index.js to do the profile extraction on redirect
-      if (err) {
-        dispatch(lockError(err))
-        return
+  const options = {
+    auth: {
+      'redirectUrl': window.location.origin + '/en-US/redirect',
```
Comment (davidascher) [inline src/actions.js]: Note that we will need to fix this when we deal with localization.

**Pull request merged**